### PR TITLE
Updated the install script to work on newer versions of ubuntu and debian…

### PR DESCRIPTION
**Pull request by avsdev-cw**
Works for debian buster & bullseye, ubuntu 18.04 and 20.04

### CODE REVIEW

File: examples/install_with_clone.sh [hunk 1]
```diff
@@ -71,34 +90,26 @@ mysql < sql/0.5.0.sql
...
-a2enmod proxy_http
-a2enmod proxy_balancer
-
 # Enable site
 a2ensite lxd_manager
 
-systemctl restart apache2
+systemctl restart apache2 || exit $?
 
+IP=`ip route get 1 | sed -n 's/^.*src \([0-9.]*\) .*$/\1/p'`
```
Comment (turtle0x1) [inline examples/install_with_clone.sh:112]: That's naughty - i like it

**Comment by turtle0x1** [comment]
This is better, thank you

The only thing I could find with ease on `|| exit $?` was [this answer on stackoverflow ](https://stackoverflow.com/a/1379904/4008082) which says to just use `set -e` is that not better than all the additional checks ?

File: examples/install_with_clone.sh [hunk 2]
```diff
@@ -71,34 +90,26 @@ mysql < sql/0.5.0.sql
...
 
-pm2 startup
+pm2 startup || exit $?
 
-pm2 save
+pm2 save || exit $?
 
 # Add cron job for gathering data
-crontab -l | { cat; echo "*/5 * * * * php /var/www/LxdMosaic/src/cronJobs/fleetAnalytics.php"; } | crontab -
+crontab -l 2>/dev/null | { cat; echo "*/5 * * * * php /var/www/LxdMosaic/src/cronJobs/fleetAnalytics.php"; } | crontab -
```
Comment (turtle0x1) [inline examples/install_with_clone.sh:104]: `2>/dev/null` why was this added to this line, but not the line below ?

Have you made sure the containers run these cron jobs ?

Comment (avsdev-cw) [reply examples/install_with_clone.sh:104]: crontab -l puts a warning on stderr when the user (root) has no crontab (which is standard for a fresh install). 2>/dev/null just hides that and it's not needed on the 2nd line because by then there is a crontab (created by the first line)

Yup, the cron tasks are install and do run. I connected each of the containers to the host LXD instance to do some quick testing as well.

I should mention that v0.8 has a couple of moderate level npm audit issues

File: examples/install_with_clone.sh (hunk 1 above)
Comment (avsdev-cw) [reply examples/install_with_clone.sh:112]: Well if you want to bind apache to an ip/interface that is not used for your default gateway then you need to edit the apache site config file anyway so...

**Comment by avsdev-cw** [comment]
> Please could you remove the file sql/users.sql

Done, it's probably worth doing something about the centos install script since it still depends on the sql/users.sql file?

**Comment by turtle0x1** [comment]
Yeah the centos, script needs updating it should only be the same changes to the `sed` & `mysql` do you mind doing them or shall I ?

File: examples/install_with_clone.sh (hunk 2 above)
Comment (turtle0x1) [reply examples/install_with_clone.sh:104]: Ah okay that makes sense! 

There is only 1 low alert on the latest branch!

**Comment by avsdev-cw** [comment]
I'm away from my dev machine now so probably best for you to do that one

**Comment by turtle0x1** [comment]
I have retested this on cetnos and ubuntu a little, will do more tommorow before tagging a release, 

@avsdev-cw as you know Debian a little better would you mind just testing on debian ? If you dont mind you will need to change  `checkout v0.9.0` to `checkout master`

**Comment by turtle0x1** [comment]
> Can you guarantee there will only be 4 instances of =, +, / per string?
> 
> The first time I ran it:
> 
> ```
> $ openssl rand -base64 29
> WKWLS+wQIqSG+JVcrovjSgm+QyUHrJ/cUiLIYaA=
> ```
> 
> might be worth making an over long string (eg 64 length) and then using the cut to bring it to a nominal length

Im not sure where your getting the limit of 4 from ?
